rust/lance/src/index/vector/flat.rs

```diff
@@ -82,6 +57,8 @@ pub async fn flat_search(
         return Ok(RecordBatch::new_empty(schema_with_distance.into()));
     }
 
+    // TODO: waiting to do this until the end adds quite a bit of latency. We should
```

Comment:
IIUC, waiting here is the existing behavior, there is no regression, right?

Reply:
Yeah, this is existing behavior. Just adding in more details from observations in tracing.